tests/testthat/test-shinytest2-reporter.R

```diff
@@ -0,0 +1,81 @@
+testthat::test_that("e2e: reporter tab is only created when a module has reporter", {
+  app_without_reporter <- TealAppDriver$new(
+    data = simple_teal_data(),
+    modules = example_module(label = "Example Module")
+  )
+  app_with_reporter <- TealAppDriver$new(
+    data = simple_teal_data(),
+    modules = report_module(label = "Module with Reporter")
+  )
+
+  teal_tabs <- app_with_reporter$get_html(selector = "#teal-main_ui-root-active_tab") %>%
+    rvest::read_html() %>%
+    rvest::html_elements("a")
+  reporter_tabs <- setNames(
+    rvest::html_attr(teal_tabs, "data-value"),
+    rvest::html_text(teal_tabs)
+  )
+  teal_tabs <- app_without_reporter$get_html(selector = "#teal-main_ui-root-active_tab") %>%
+    rvest::read_html() %>%
+    rvest::html_elements("a")
+  non_reporter_tabs <- setNames(
+    rvest::html_attr(teal_tabs, "data-value"),
+    rvest::html_text(teal_tabs)
+  )
+
+  testthat::expect_identical(
+    non_reporter_tabs,
+    c("Example Module" = "example_module")
+  )
+  testthat::expect_identical(
+    reporter_tabs,
+    c("Module with Reporter" = "module_with_reporter", "Report previewer" = "report_previewer")
+  )
+
+  app_without_reporter$stop()
+  app_with_reporter$stop()
+})
+
+testthat::test_that("e2e: adding a report card in a module adds it in the report previewer tab", {
+  app <- TealAppDriver$new(
+    data = simple_teal_data(),
+    modules = report_module(label = "Module with Reporter")
+  )
+  app$wait_for_idle(timeout = default_idle_timeout)
+
+  app$click(NS(app$active_module_ns(), "reporter-add_report_card_simple-add_report_card_button"))
```

Comment:
This introduces a coupling to the ids defined in `teal.reporter`. If we ever change it for some reason - `teal` would have to be re-released. The same goes with `teal.slice` and other packages. I would like to avoid this and keep releases as much independent as possible. 
This actually happened in the past - on `teal.slice` refactor we broke tests in `teal.modules.hermes` just because they were written like this.
Please invest in finding a good way to avoid such coupling.

Reply:
This is a difficult problem with E2E tests, in particular with `shinytest2` which has a workflow that is geared towards using IDs and CSS selectors.

We could mitigate some of this coupling by adding helpers that identify elements from labels/content, but there will always be some dependency on IDs / selectors / labels.

Maybe we could use a CSS selector pattern that looks for "ends with" `parent-context [id$=-add_report_card_button]` instead of relying on the full namespace.

Reply:
What if we test that only in `{teal.reporter}`, even though we have a class in `{teal}`?

Reply:
`{teal.reporter}` would need to have `{teal}` on Suggests or we would need to implement a non-standard way of testing (making use of staged dependencies environment).

It may be possible, but the question becomes: "should we?"

Reply:
The idea of a testing E2E package is growing on me. One that would contain:
- TealAppDriver
- E2E tests for `teal.{code,data,slices,reporter,widgets} packages.

All packages that already depend on `teal` could have the e2e tests living on them.

Reply:
yeah, but it's hard to decide what to do with packages on which `{teal}` relies. At least half of the problem is solved :P

Reply:
Thanks guys for the review.
I created a new issue #1160 so we can move forward with this PR.